posthog/models/feature_flag/flag_matching.py

```diff
@@ -299,39 +301,47 @@ def query_conditions(self) -> Dict[str, bool]:
 
-                            # TRICKY: Due to property overrides, we sometimes shortcircuit the condition check.
-                            # In that case, the expression is empty, and we can assume the condition is a match.
-                            if not expr:
-                                all_conditions[key] = True
-
-                        if feature_flag.aggregation_group_type_index is None:
-                            person_query = person_query.annotate(
-                                **{
-                                    key: ExpressionWrapper(
-                                        expr if expr else RawSQL("true", []), output_field=BooleanField()
-                                    )
-                                }
-                            )
-                            person_fields.append(key)
-                        else:
-                            if feature_flag.aggregation_group_type_index not in group_query_per_group_type_mapping:
-                                # ignore flags that didn't have the right groups passed in
-                                continue
-                            group_query, group_fields = group_query_per_group_type_mapping[
-                                feature_flag.aggregation_group_type_index
-                            ]
-                            group_query = group_query.annotate(
-                                **{
-                                    key: ExpressionWrapper(
-                                        expr if expr else RawSQL("true", []), output_field=BooleanField()
-                                    )
-                                }
-                            )
-                            group_fields.append(key)
-                            group_query_per_group_type_mapping[feature_flag.aggregation_group_type_index] = (
-                                group_query,
-                                group_fields,
-                            )
+                            # TRICKY: Due to property overrides for cohorts, we sometimes shortcircuit the condition check.
+                            # In that case, the expression is either an explicit True or explicit False, or multiple conditions.
+                            # We can skip going to the database in explicit True|False conditions. This is important
+                            # as it allows resolving flags correctly for non-ingested persons.
+                            # However, this doesn't work for the multiple condition case (when expr has multiple Q objects),
+                            # but it's better than nothing.
+                            # TODO: A proper fix would be to handle cohorts with property overrides before we get to this point.
+                            # Unskip test test_complex_cohort_filter_with_override_properties when we fix this.
+                            if expr == Q(pk__isnull=False) or expr == Q(pk__isnull=True):
```

Comment:
NIT: maybe this is more readable?

```
if expr == Q(pk__isnull=False):
...
if expr == Q(pk__isnull=True):
...
```

Reply:
nice, it is indeed.

Will make the update in the follow up decide speedup